magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md

```diff
@@ -0,0 +1,43 @@
+# Fixing a bad deploy
+
+Normally your Auto-Scaling Group will be  configured with a maximum capacity that is _twice_ the 'desired'
+capacity - this is to allow deploys to work by _doubling_ the desired capacity, bringing up new boxes, before
+killing the old ones off.
+
+When a Riff-Raff/Magenta deploy goes bad, your Auto-Scaling Group will often be left in a 'bad' state, where
+the desired capacity stays the _same_ as the specified maximum capacity for the ASG - once in this
+state, Riff-Raff/Magenta will not be able to do any more automated deploys, because it can't double the desired
+capacity in excess of your configured max.
+
+You need to manually fix this problem, and this document gives some possible approaches for doing it.
+
+Once the ASG is back to it's normal size, do another automated deploy to verify that you can still
+deploy successfully, and that all live boxes are in the correct state.
+
+
+## A) Manually Kill Bad Boxes
+
+Use the AWS CLI [`terminate-instance-in-auto-scaling-group`](http://docs.aws.amazon.com/cli/latest/reference/autoscaling/terminate-instance-in-auto-scaling-group.html)
+command to kill bad boxes by instance-id, making sure you leave good (old) boxes still operational:
+
+```
+$ aws autoscaling terminate-instance-in-auto-scaling-group --should-decrement-desired-capacity --instance-id i-badbadba
```

Comment:
I'd put a marker in the command, or a comment, or both underlining that a real instance id should be provided